python/ray/_private/utils.py

```diff
@@ -1554,3 +1555,48 @@ def split_address(address: str) -> Tuple[str, str]:
 
     module_string, inner_address = address.split("://", maxsplit=1)
     return (module_string, inner_address)
+
+
+def get_or_create_event_loop() -> asyncio.BaseEventLoop:
+    """Get a running async event loop if one exists, otherwise create one.
+
+    This function serves as a proxy for the deprecating get_event_loop().
+    It tries to get the running loop first, and if no running loop
+    could be retrieved:
+    - For python version <3.10: it falls back to the get_event_loop
+        call.
+    - For python version >= 3.10: it uses the same python implementation
+        of _get_event_loop() at asyncio/events.py.
+
+    Ideally, one should use high level APIs like asyncio.run() with python
+    version >= 3.7, if not possible, one should create and manage the event
+    loops explicitly.
+    """
+    loop = None
+    try:
+        loop = asyncio.get_running_loop()
+    except RuntimeError:
+        pass
+
+    if loop is not None:
+        return loop
+
+    import sys
+
+    vers_info = sys.version_info
+
+    # python 2.7 and below doesn't have `major` and `minor` fields.
+    if sys.version_info[0] == 2:
+        return asyncio.get_event_loop()
+
+    if vers_info.major >= 3 and vers_info.minor >= 10:
```

Comment:
Should we specialize 3.6 instead of 3.10 given we will remove support for 3.6? Like calling `asyncio.get_event_loop_policy().get_event_loop()` if >=3.7?